include/libopencm3/pac55xx/gpio.h

```diff
@@ -0,0 +1,367 @@
+#define GPIOA_INTEDGEBOTHR          GPIO_INTEDGEBOTHR(GPIOA)
+#define GPIOB_INTEDGEBOTHR          GPIO_INTEDGEBOTHR(GPIOB)
+#define GPIOC_INTEDGEBOTHR          GPIO_INTEDGEBOTHR(GPIOC)
+#define GPIOD_INTEDGEBOTHR          GPIO_INTEDGEBOTHR(GPIOD)
+#define GPIOE_INTEDGEBOTHR          GPIO_INTEDGEBOTHR(GPIOE)
+#define GPIOF_INTEDGEBOTHR          GPIO_INTEDGEBOTHR(GPIOF)
+#define GPIOG_INTEDGEBOTHR          GPIO_INTEDGEBOTHR(GPIOG)
+/**@}*/
+
+/* GPIO Clock Synchronization Settings. When set, this enables 3-clock synchronizer on pins. */
+/** @defgroup gpio_clksync_regs GPIO CLKSYNC register mapping.
+@ingroup PAC55xx_gpio
+@{*/
+#define GPIO_CLKSYNCR(base)         MMIO32((base) + 0x2C)
+#define GPIOA_CLKSYNCR              GPIO_CLKSYNCR(GPIOA)
+#define GPIOB_CLKSYNCR              GPIO_CLKSYNCR(GPIOB)
+#define GPIOC_CLKSYNCR              GPIO_CLKSYNCR(GPIOC)
+#define GPIOD_CLKSYNCR              GPIO_CLKSYNCR(GPIOD)
+#define GPIOE_CLKSYNCR              GPIO_CLKSYNCR(GPIOE)
+#define GPIOF_CLKSYNCR              GPIO_CLKSYNCR(GPIOF)
+#define GPIOG_CLKSYNCR              GPIO_CLKSYNCR(GPIOG)
+/**@}*/
+
+/* GPIO Set Register. This register can be used for atomic setting of outputs. */
+/** @defgroup gpio_doset_regs GPIO DOSET register mapping.
+@ingroup PAC55xx_gpio
+@{*/
+#define GPIO_DOSETR(base)           MMIO32((base) + 0x30)
+#define GPIOA_DOSETR                GPIO_DOSETR(GPIOA)
+#define GPIOB_DOSETR                GPIO_DOSETR(GPIOB)
+#define GPIOC_DOSETR                GPIO_DOSETR(GPIOC)
+#define GPIOD_DOSETR                GPIO_DOSETR(GPIOD)
+#define GPIOE_DOSETR                GPIO_DOSETR(GPIOE)
+#define GPIOF_DOSETR                GPIO_DOSETR(GPIOF)
+#define GPIOG_DOSETR                GPIO_DOSETR(GPIOG)
+/**@}*/
+
+/* GPIO Set Register. This register can be used for atomic setting of outputs. */
+/** @defgroup gpio_doclear_regs GPIO DOCLEAR register mapping.
+@ingroup PAC55xx_gpio
+@{*/
+#define GPIO_DOCLEARR(base)         MMIO32((base) + 0x34)
+#define GPIOA_DOCLEARR              GPIO_DOCLEARR(GPIOA)
+#define GPIOB_DOCLEARR              GPIO_DOCLEARR(GPIOB)
+#define GPIOC_DOCLEARR              GPIO_DOCLEARR(GPIOC)
+#define GPIOD_DOCLEARR              GPIO_DOCLEARR(GPIOD)
+#define GPIOE_DOCLEARR              GPIO_DOCLEARR(GPIOE)
+#define GPIOF_DOCLEARR              GPIO_DOCLEARR(GPIOF)
+#define GPIOG_DOCLEARR              GPIO_DOCLEARR(GPIOG)
+/**@}*/
+
+/* GPIO Overlay Structures (for convenience). */
+/** @defgroup gpio_overlay_regs GPIO Struct Overlay.
+@ingroup PAC55xx_gpio
+@{*/
+typedef struct {
+  volatile uint32_t mode;
+  volatile uint32_t outmask;
+  volatile uint32_t out;
+  volatile const uint32_t in;
+  volatile uint32_t inten;
+  volatile const uint32_t intflag;
+  volatile uint32_t intclear;
+  volatile uint32_t inttype;
+  volatile uint32_t intcfg;
+  volatile uint32_t intintedgeboth;
+  volatile uint32_t clksync;
+  volatile uint32_t doset;
+  volatile uint32_t doclear;
+} pac55xx_gpio_regs_t;
```

Comment:
providing both styles?  are you going to be maintaining two apis?  so far we've _only_ done define style.  struct style might often be considered nicer, but changing things is.... not good.